astropy_timeseries/sampled.py

```diff
@@ -161,22 +171,38 @@ def from_pandas(self, df):
         # TODO: determine how user can specify time scale
         time = Time(df.index)
 
-        # Create table without the time column
-        table = Table.from_pandas(df)
+        if LooseVersion(astropy.__version__) > LooseVersion("3.1.1"):
+            table = Table.from_pandas(df, index=index)
+        else:
+            table = Table.from_pandas(df)
 
         return TimeSeries(time=time, data=table)
 
-    def to_pandas(self):
+    def to_pandas(self, index=None):
         """
-        Convert this time series to a :class:`~pandas.DataFrame` with a
-        :class:`~pandas.DatetimeIndex` index.
+        Convert this :class:`~astropy_timeseries.TimeSeries` to a
+        :class:`~pandas.DataFrame` with a :class:`~pandas.DatetimeIndex` index.
+
+        Parameters
+        ----------
+        index : None, bool, str
+            Specify DataFrame index mode
+            Only used if Astropy version >=``3.2``.
+
+        Returns
+        -------
+        dataframe : :class:`pandas.DataFrame`
+            A pandas :class:`pandas.DataFrame` instance
         """
 
         # Extract table without time column
         table = self[[x for x in self.colnames if x != 'time']]
 
         # First make a normal pandas dataframe
-        df = Table(table).to_pandas()
+        if LooseVersion(astropy.__version__) > LooseVersion("3.1.1"):
```

Comment:
As an alternative to checking the version, you could inspect the function signature?

Reply:
Let's just check the version with an ``ASTROPY_LT_32`` constant - once we merge this into astropy we can more easily search and remove then.